src/Amazon.AspNetCore.DataProtection.SSM/XmlDeletableKeyManager.cs

```diff
@@ -0,0 +1,90 @@
+﻿/*
+Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  A copy of the License is located at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  or in the "license" file accompanying this file. This file is distributed
+  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+  express or implied. See the License for the specific language governing
+  permissions and limitations under the License.
+ */
+
+using Microsoft.AspNetCore.DataProtection.Internal;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Amazon.AspNetCore.DataProtection.SSM
+{
+#if NET9_0_OR_GREATER
+    /// <summary>
+    /// Extension of XmlKeyManager which supports key deletion.
+    /// 
+    /// <para>
+    /// XmlKeyManager implements IKeyManager but doesn't have the declaration for implementing IDeletableKeyManager. 
+    /// We created our wrapper to have the IDeletableKeyManager declaration. Refer https://github.com/dotnet/aspnetcore/discussions/60315 for more details.
+    /// </para>
+    /// </summary>
+    public class XmlDeletableKeyManager : IDeletableKeyManager
```

Comment:
This class should be `internal`. We don't want users depending on this class.

Reply:
Fixed.